sponsorship/src/main/java/com/kyriba/conference/sponsorship/service/PlanService.java

```diff
@@ -1,14 +1,52 @@
 package com.kyriba.conference.sponsorship.service;
 
+import com.kyriba.conference.sponsorship.dao.PlanRepository;
+import com.kyriba.conference.sponsorship.dao.SponsorRepository;
+import com.kyriba.conference.sponsorship.domain.Plan;
+import com.kyriba.conference.sponsorship.domain.PlanCategory;
+import com.kyriba.conference.sponsorship.domain.Sponsor;
+import com.kyriba.conference.sponsorship.domain.dto.PlanDto;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import javax.transaction.Transactional;
 
 
 /**
  * @author M-ASL
  * @since v1.0
  */
 @Service
+@Transactional
+@RequiredArgsConstructor
 public class PlanService

```

Comment:
Usually Service is defined as an intefrace and then implementation is added as ServiceImpl.

Reply:
Why is it needed? Won't it be redundant, because there is only one single implementation of the service?

Reply:
It is needed to specify available API and split declaration from implemetation details.

Reply:
Ok, I'll introduce an interface for the service. And as far as I remember, you told us that we should avoid using default methods in such service interfaces to avoid implementation details at declaration side, right? Please, correct me if I'm wrong.

Reply:
Good note regarding default service methods: as I know Transactional annotation won't work for such a method. But maybe Spring can resolve this case? Who knows?

Reply:
Not to miss the question in a thread :)
@xpinjection please, see my question above

Reply:
No default methods on service interface. :)